python/packages/autogen-ext/src/autogen_ext/models/anthropic/_anthropic_client.py

```diff
@@ -503,19 +502,12 @@ async def create(
             completion_tokens=result.usage.output_tokens,
         )
 
-        # Log the event if in a handler context
-        try:
-            agent_id = MessageHandlerContext.agent_id()
-        except RuntimeError:
-            agent_id = None
-
         logger.info(
             LLMCallEvent(
                 messages=cast(Dict[str, Any], anthropic_messages),
                 response=result.model_dump(),
                 prompt_tokens=usage.prompt_tokens,
                 completion_tokens=usage.completion_tokens,
-                agent_id=agent_id,
```

Comment:
It seems that what is happening here is that the agent_id is added at the logging library level instead of having the model client add this?
(which is great!! single point of failure :))

Reply:
Yep, just refactored it as I see it.